sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/Utility.java

```diff
@@ -706,4 +725,204 @@ public static int getDocumentCount(Iterable<?> documents) {
+            SentenceOpinionPropertiesHelper.setAssessments(sentenceOpinion, new IterableStream<>(assessmentSentiments));
+            sentenceOpinions.add(sentenceOpinion);
+        });
+
+        return new IterableStream<>(sentenceOpinions);
+    }
+
+    /*
+     * Transform type TargetConfidenceScoreLabel to SentimentConfidenceScores.
+     */
+    private static SentimentConfidenceScores toSentimentConfidenceScores(
+        TargetConfidenceScoreLabel targetConfidenceScoreLabel) {
+        return new SentimentConfidenceScores(targetConfidenceScoreLabel.getNegative(), NEUTRAL_SCORE_ZERO,
+            targetConfidenceScoreLabel.getPositive());
+    }
+
+    /*
+     * Transform type SentenceOpinion to OpinionSentiment.
+     */
+    private static AssessmentSentiment toAssessmentSentiment(SentenceAssessment sentenceAssessment) {
+        final AssessmentSentiment assessmentSentiment = new AssessmentSentiment();
+        AssessmentSentimentPropertiesHelper.setText(assessmentSentiment, sentenceAssessment.getText());
+        AssessmentSentimentPropertiesHelper.setSentiment(assessmentSentiment,
+            TextSentiment.fromString(sentenceAssessment.getSentiment().toString()));
+        AssessmentSentimentPropertiesHelper.setConfidenceScores(assessmentSentiment,
+            toSentimentConfidenceScores(sentenceAssessment.getConfidenceScores()));
+        AssessmentSentimentPropertiesHelper.setNegated(assessmentSentiment, sentenceAssessment.isNegated());
+        AssessmentSentimentPropertiesHelper.setOffset(assessmentSentiment, sentenceAssessment.getOffset());
+        AssessmentSentimentPropertiesHelper.setLength(assessmentSentiment, sentenceAssessment.getLength());
+        return assessmentSentiment;
+    }
+
+    /*
+     * Parses the reference pointer to an index array that contains document, sentence, and opinion indexes.
+     */
+    public static int[] parseRefPointerToIndexArray(String assessmentPointer) {
+        // The pattern always start with character '#', the assessment index will existing in specified sentence, which
+        // is under specified document.
+        // example: #/documents/0/sentences/0/assessments/0
+        final String patternRegex = "#/documents/(\\d+)/sentences/(\\d+)/assessments/(\\d+)";
+        final Pattern pattern = Pattern.compile(patternRegex);
+        final Matcher matcher = pattern.matcher(assessmentPointer);
+        final boolean isMatched = matcher.find();
+
+        // The first index represents the document index, second one represents the sentence index,
+        // third ond represents the assessment index.
+        final int[] result = new int[3];
+
+        if (isMatched) {
+            String[] segments = assessmentPointer.split("/");
+            result[0] = Integer.parseInt(segments[2]);
+            result[1] = Integer.parseInt(segments[4]);
+            result[2] = Integer.parseInt(segments[6]);
```

Comment:
No need to do this splitting logic, you're already capturing the numbers in your `Matcher` above.

```java
result[0] = Integer.parseInt(matcher.group(1));
result[1] = Integer.parseInt(matcher.group(2));
result[2] = Integer.parseInt(matcher.group(3));
```